### ISSUE TITLE
[feature] Support standard URLs in backup-source options

### ISSUE THREAD

**Issue by klimkin** [conan-io/conan#14341]
### What is your suggestion?

Option values for `core.sources:download_urls` and `core.sources:upload_url` should allow standard URLs . Feels like Conan2 is using str join/format instead of url join.

This works:
```
core.sources:download_urls=["https://artifactory.com/sources-backup/", "origin"]
core.sources:upload_url=https://artifactory.com/sources-backup/
```

This does not:
```
core.sources:download_urls=["https://artifactory.com/sources-backup", "origin"]
core.sources:upload_url=https://artifactory.com/sources-backup
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Thanks for reporting this @klimkin 

Proposing https://github.com/conan-io/conan/pull/14342 for next 2.0.10 release.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14342 merged, it will be in next 2.0.10
